Session.cc

```diff
@@ -1243,7 +1283,7 @@ void Session::SendEvent(CARTA::EventType event_type, uint32_t event_id, google::
     head->icd_version = carta::ICD_VERSION;
     head->request_id = event_id;
     message.SerializeToArray(msg.data() + sizeof(carta::EventHeader), message_length);
-    _out_msgs.push(msg);
+    _out_msgs.push(std::make_pair(msg, compress));
```

Comment:
doesn't this result in an additional copy? you should rather form the pair first, and then serialize to the pair's `msg.data()`